libcloud/common/durabledns.py

```diff
@@ -0,0 +1,329 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from bs4 import BeautifulSoup
+
+from libcloud.utils.py3 import PY3, b
+
+from libcloud.utils.misc import lowercase_keys
+from libcloud.common.base import ConnectionUserAndKey
+from libcloud.common.base import XmlResponse
+
+
+# API HOST to connect
+API_HOST = 'durabledns.com'
+
+# REQUEST SCHEMAS
+LIST_ZONES = """
```

Comment:
Please use `ElementTree` module to construct XML documents (see EC2, S3 and other XML based drivers for an example).

Creating XML by manually concatenating strings is not safe nor robust.